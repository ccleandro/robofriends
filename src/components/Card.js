import React from "react";
//const Card = props => {
const Card = ({ id, name, email }) => {
  return (
    <div className="tc bg-light-green dib br3 pa3 ma2 grow ba b--black-10 shadow-5 ">
      <img src={`https://robohash.org//${id}?200x200`} alt="robots"></img>
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
