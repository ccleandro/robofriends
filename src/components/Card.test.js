// shallow - ONLY THE COMPONENT, UNIT ensure that your tests aren't indirectly asserting on behavior of child components
// mount - FULL DOM mounts the component in the DOM, it must be run in an environment that at least “looks like” a browser environment - jsdom
// render - STATIC generate HTML from your React tree, and analyze the resulting HTML structure. uses a third party HTML parsing and traversal library Cheerio
import { shallow } from "enzyme";
import React from "react";
import Card from "./Card";

it("expect to render 1 Card component", () => {
  expect(shallow(<Card />).length).toEqual(1);
});

it("to match the snapshot", () => {
  expect(shallow(<Card />)).toMatchSnapshot();
});
