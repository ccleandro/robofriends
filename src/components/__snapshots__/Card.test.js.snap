// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match the snapshot 1`] = `ShallowWrapper {}`;
